src/foam/nanos/auth/CountryService.java

```diff
@@ -0,0 +1,86 @@
+package foam.nanos.auth;
+
+import foam.core.ContextAwareSupport;
+import foam.dao.ArraySink;
+import foam.dao.DAO;
+import foam.mlang.MLang;
+import foam.mlang.predicate.Predicate;
+import foam.nanos.NanoService;
+import foam.nanos.notification.email.EmailMessage;
+import foam.nanos.notification.email.EmailService;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Query the country by code, name, or alternative names.
+ */
+public class CountryService extends ContextAwareSupport implements NanoService {
```

Comment:
Separate email from look. Lookup can just be a static method rather than a service since its stateless.